sklearn/utils/estimator_checks.py

```diff
@@ -1191,6 +1192,58 @@ def check_estimators_pickle(name, estimator_orig):
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
 
+def check_pairwise_estimator_tag(name, estimator_orig):
+    attributes_to_check = [('metric', 'precomputed'),
+                           ('affinity', 'precomputed'),
+                           ('kernel', 'precomputed')]
+
+    # Check if _pairwise attribute is present - will be used later
+    has_pairwise_tag = (False
+                        if getattr(estimator_orig, '_pairwise', None) is None
+                        else True)
+
+    # Using iris as sample data
+    iris = load_iris()
+    X, y_ = iris.data, iris.target
+    distance_matrix = pairwise_distances(X)
+
+    for attribute, attribute_value in attributes_to_check:
+        # Check to see if attribute value is supported by estimator
+        if attribute not in estimator_orig.get_params():
+            continue
+        try:
+            # Construct new object of estimator with desired attribute value
+            modified_estimator = clone(estimator_orig).set_params(
+                                                **{attribute: attribute_value})
+            # Not all estimators validate parameters, so check fit()
+            modified_estimator.fit(distance_matrix, y_)
+        except (TypeError, ValueError, KeyError):
```

Comment:
I don't see the relevance of KeyError.

Reply:
So Nystroem must not currently support precomputed... We could:
* add support there
* Just catch Exception here

Reply:
I’m not sure if I’m understanding this correctly, but the exception handling includes KeyError in order to handle Nystroem (since Nystroem raises a KeyError instead of a TypeError/ValueError)

Reply:
If you just catch Exception here, rather than `ValueError, TypeError, KeyError` I think that would be better.